### PR TITLE
Implemented loading LocalSettings.loadModulesOnStartup

### PR DESCRIPTION
**Pull request by richATavail**
- Implemented loading `LocalSettings.loadModulesOnStartup`
- Re-abstracted `AbstractWorkbenchTask` to be independent of a `ResolvedModuleName`, opting to create a subtype of `AbstractWorkbenchTask`, `AbstractWorkbenchModuleTask`, that works specifically with non-nullable `ResolvedModuleNames`.
- Now from context menu you can add or remove a module for load on anvil start up.

### CODE REVIEW

**Comment by markATavail** [comment]
During a quick smoke test, I saw that if you select a module/package and then "Add Load on Anvil Start", then bring up the menu again, that menu item is still enabled, and "Remove Load on Anvil Start" is still disabled.  Changing the selection then changing it back caused the menu items to be updated correctly.

**Comment by richATavail** [comment]
> During a quick smoke test, I saw that if you select a module/package and then "Add Load on Anvil Start", then bring up the menu again, that menu item is still enabled, and "Remove Load on Anvil Start" is still disabled. Changing the selection then changing it back caused the menu items to be updated correctly.


@markATavail I identified the issue and have fixed it. When not navigating away from the selected module after changing the load state on start, the context menu doesn't get rebuilt to represent the new state. This required calling `AvailWorkbench.setEnablements` after each change to force the context menu to be rebuilt without having to choose a different module.

**Comment by richATavail** [comment]
@markATavail Now there is only one menu option that changes based on whether or not the module has been added to local settings.
image
image

File: avail/src/main/kotlin/avail/anvil/tasks/BuildManyTask.kt [hunk 1]
```diff
@@ -30,46 +30,45 @@
...
 import avail.anvil.AvailWorkbench
 import avail.builder.ResolvedModuleName
-import org.availlang.artifact.environment.project.LocalSettings
-import java.awt.event.ActionEvent
-import javax.swing.Action
 
 /**
- * An [AbstractWorkbenchAction] that adds the [ResolvedModuleName] to the
- * [LocalSettings.loadModulesOnStartup].
+ * An [AbstractBuildTask] that launches the actual build of multiple
```
Comment (markATavail) [inline avail/src/main/kotlin/avail/anvil/tasks/BuildManyTask.kt:39]: Comment has cloning errors.

File: avail/src/main/kotlin/avail/anvil/tasks/DocumentationTask.kt [hunk 2]
```diff
@@ -35,12 +35,12 @@ package avail.anvil.tasks
...
 import avail.descriptor.module.ModuleDescriptor
 import avail.anvil.AvailWorkbench
-import avail.anvil.AvailWorkbench.AbstractWorkbenchTask
+import avail.anvil.AvailWorkbench.AbstractWorkbenchModuleTask
 import java.awt.Cursor
 
 
 /**
- * A `DocumentationTask` initiates and manages documentation
+ * An [AbstractWorkbenchModuleTask] that initiates and manages documentation
```
Comment (markATavail) [inline avail/src/main/kotlin/avail/anvil/tasks/DocumentationTask.kt:43]: Update class name in comment.

**Review by markATavail — APPROVED** [review]

File: avail/src/main/kotlin/avail/anvil/AvailWorkbench.kt [hunk 3]
```diff
@@ -1052,6 +1045,54 @@ class AvailWorkbench internal constructor(
...
+	 *
+	 * @constructor
+	 * Construct a new `AbstractWorkbenchModuleTask`.
+	 *
+	 * @param workbench
+	 *   The owning [AvailWorkbench].
+	 * @param targetModuleName
+	 *   The resolved name of the target [module][ModuleDescriptor].
+	 */
+	abstract class AbstractWorkbenchModuleTask constructor(
```
[the remark this answers is not in the document]
Comment (richATavail) [reply avail/src/main/kotlin/avail/anvil/AvailWorkbench.kt]: Done.

**Comment by richATavail** [comment]
@markATavail Now there is only one menu option that changes based on whether or not the module has been added to local settings.
image

**Pull request merged**